src/boxStorage.ts

```diff
@@ -1,11 +1,10 @@
-import { EncodedBoxReference } from './types/transactions/index.js';
 import { BoxReference } from './types/transactions/base.js';
 
-function translateBoxReference(
+function boxReferenceToEncodingData(
   reference: BoxReference,
   foreignApps: bigint[],
   appIndex: bigint
-): EncodedBoxReference {
+): Map<string, unknown> {
```

Comment:
What was rationale for not going with an alias for the types both here and on `translateBoxReference`?

Reply:
This PR removes many named object types like `EncodedBoxReference` and `EncodedTransaction` which were used as the intermediate representation before JSON/msgpack encoding.

The reason being that we've upgraded from objects to maps to support non-string keys. And with maps, there's no convenient way to create a type alias. I actually think this change is good, as before the named object types were used in many places where they shouldn't have been, and their typed nature implied that certain fields were checked/correctly typed when in many cases we casted directly to those types after decoding with no additional checks.